plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitManager.java

```diff
@@ -104,9 +105,30 @@ public ConnectorSplitSource getSplits(
         TupleDomain<ColumnHandle> tableConstraint = bigQueryTableHandle.constraint();
         Optional<String> filter = BigQueryFilterQueryBuilder.buildFilter(tableConstraint);
 
-        if (!bigQueryTableHandle.isNamedRelation()) {
+        if (bigQueryTableHandle.isQueryRelation()) {
+            BigQueryQueryRelationHandle bigQueryQueryRelationHandle = bigQueryTableHandle.getRequiredQueryRelation();
             List<BigQueryColumnHandle> columns = bigQueryTableHandle.projectedColumns().orElse(ImmutableList.of());
-            return new FixedSplitSource(BigQuerySplit.forViewStream(columns, filter));
+            boolean useStorageApi = bigQueryQueryRelationHandle.isUseStorageApi();
+            List<String> projectedColumnsNames = getProjectedColumnNames(columns);
+
+            String query = filter
+                    .map(whereClause -> "SELECT " + String.join(",", projectedColumnsNames) + " FROM (" + bigQueryQueryRelationHandle.getQuery() + ") WHERE " + whereClause)
```

Comment:
> String.join(",", projectedColumnsNames)

This may create a invalid query `SELECT FROM (...)` if projectedColumnsNames is an empty right? I'm not sure if such situations really happen though. Can we change to `*` or throw an exception? 

Also, why is `projectedColumnsNames` unused when `filter` doesn't exist?